bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -703,7 +716,7 @@ void stop() {
      *          Entry Log File Id
      */
     protected boolean removeEntryLog(long entryLogId) {
-        removeFromChannelsAndClose(entryLogId);
+        fileChannelBackingCache.removeFromChannelsAndClose(entryLogId);
```

Comment:
What about the entries still in the caches? I would suggest you get rid of ```removeFromChannelsAndClose``` completely, and invalidate the entry from all the caches. You could also do nothing, as the cache would release their refcount eventually, but better to help it along a little.

Reply:
We can't invalidate it from all the caches eg. logid2ReadChannel as it's a threadLocal variable. Do you think it's okay to delete  the entryLog file before close the corresponding fileChannel? I'm not sure if the JVM will throw exception if we do that. If it's fine then we can remove it at all.

Reply:
I guess it would throw an exception when the file is eventually dropped from all caches and it tries to close it. This could be unpredictable, as removals happen when another action trigger it, like a put, and it's that put that would get the exception.

Reply:
After add a global readChannel concurrent list, we can let invalidating cache to trigger close operation. So is it still a need to keep the removeFromChannelsAndClose()? I guess it's not anymore.